scripts/download-resources.mjs

```diff
@@ -80,53 +90,173 @@ export async function download(url, destPath, overwrite = false, access = fs.con
   }
 }
 
-export default async function main() {
-  const resourcesDir = path.join(process.cwd(), 'resources', os.platform());
-  const binDir = path.join(resourcesDir, 'bin');
-  /** The platform string, as used by golang / Kubernetes. */
-  const kubePlatform = {
-    darwin: 'darwin',
-    linux:  'linux',
-    win32:  'windows',
-  }[os.platform()];
-
-  fs.mkdirSync(binDir, { recursive: true });
-
-  // Download Kubectl
-  const kubeVersion = '1.20.7';
-  const kubectlURL = `https://dl.k8s.io/v${ kubeVersion }/bin/${ kubePlatform }/amd64/${ exeName('kubectl') }`;
-  const kubectlPath = path.join(binDir, exeName('kubectl'));
-
-  await download(kubectlURL, kubectlPath, false, fs.constants.X_OK);
-
-  // Download Helm. It is a tar.gz file that needs to be expanded and file moved.
-  const helmVersion = '3.6.0';
-  const helmURL = `https://get.helm.sh/helm-v${ helmVersion }-${ kubePlatform }-amd64.tar.gz`;
-  const helmDir = fs.mkdtempSync(path.join(os.tmpdir(), 'rd-helm-'));
+// Download a tar.gz file to a temp dir, expand,
+// and move the expected binary to the final dir
+async function downloadTarGZ(url, binaryBasename, tgzPlatformDir) {
+  const tgzDir = fs.mkdtempSync(path.join(os.tmpdir(), `rd-${ binaryBasename }-`));
+  let binaryFinalPath = '';
```

Comment:
Why not set this initially as a constant? As far as I can tell all the inputs are fixed at this point.

Reply:
It might make sense to do an overwrite check (to avoid re-downloading the file).

Reply:
It's not a constant because it's set to a new value in the `try` block.

Reply:
Umm, but either 1) The `try` succeeds, we have the value from the `try` block, or 2) the `try` throws, we never look at the value (because it's a `try`/`finally`, no `catch`, the exception is re-thrown), right?